### ISSUE TITLE
Drafts are not being automatically excluded

### ISSUE THREAD

**Issue by eduardosada** [nuxt/content#1523]
### Environment

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0-rc.9`
- Nitro Version:    `0.5.2`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `target`, `typescript`, `modules`, `runtimeConfig`
- Runtime Modules:  `@nuxt/content@2.1.0`
- Build Modules:    `-`
------------------------------

### Reproduction

https://stackblitz.com/edit/nuxt-starter-aac1e1?file=pages%2Findex.vue

run:

```bash
npm run build
npm run preview
```

### Describe the bug

https://content.nuxtjs.org/guide/writing/markdown

Docs says:

```
draft | Boolean | false | Mark the page as draft (and only display it in development mode).
```

But that property doesn't work. I saw from the code that if you put the word 'draft' at the end of a file, it sets the internal `_draft` property to true. But that doesn't make anything.

eg.:

```
test.draft.md --> marked as a draft
```

Currently all drafts are displayed in development and production mode.

**Comment by Ratismal** [comment]
Want to give my perspective on this. There's some confused reactions, so hopefully this will clarify things.

The documentation advertises a `draft` property. Assumedly, if a piece of content is marked as a draft, it should be _visible_ when viewing the site in development mode, but should be _excluded_ when the site is viewed in production mode.

A use case for this would be, for example, if content authors want to write and commit drafts of content without it being visible to the public if the site is deployed before it's finished.

Another use case would be to queue up content that should is ready to be released, but shouldn't be publicly accessible yet due to a release schedule. This is my personal use case - I'd like to be able to proofread content in development mode but I don't want things to release early.

The documentation is misleading, as this functionality doesn't actually exist in any capacity.

First of all, the `draft` property _is_ a reserved keyword, as seen here:
- https://github.com/nuxt/content/blob/main/src/runtime/utils/query.ts#L74-L97

If the property `draft` is set in the front-matter of content, it cannot even be queried against, as the field is reserved and gets excluded from queries.

It's possible that this is actually a bug - looking at the [ContentRenderer.vue](https://github.com/nuxt/content/blob/main/test/fixtures/basic/pages/features/slotted-content-renderer.vue#L19) component, `draft` _is_ used in a query! However, since it's a reserved field, it has no effect whatsoever.

Every other draft reference to drafts in the codebase are to the `_draft` property. One would assume that setting the `draft` property would set the internal `_draft` property which _can_ be queried, but this isn't the case. The only way to set the `_draft` property would be to set it directly in front-matter, or to suffix a filename with `.draft` (ex. `file-name.draft.md`).

And unfortunately, other than being set, there is nothing in the codebase that actually takes action with the `_draft` property, so ultimately it's pointless.

--- 

As far as solutions go, the cleanest thing I can think of would be to exclude content altogether during the nitro phase.
- https://github.com/nuxt/content/blob/main/src/runtime/server/storage.ts#L157 - A draft check could be performed here, to prevent the content from being cached
- https://github.com/nuxt/content/blob/main/src/runtime/server/storage.ts#L107 - A filter could be added here, to strip out any drafts from the list

I think it would be fairly easy for me to create a PR to accomplish both of these tasks, but I would like feedback about whether or not this would be a good solution before doing so.

As a workaround, I've created a plugin `server/plugins/draft.ts` that mutilates draft content when on production mode.
```ts
export default defineNitroPlugin((nitroApp) => {
  nitroApp.hooks.hook('content:file:afterParse', (file) => {
    if (file._draft && process.env.NODE_ENV === 'production') {
      console.log('Mutilating draft file', file._id);
      file._id = 'content:_:draft';
      file.body = {};
      file.title = 'Draft';
      file._path = '/draft';
      file._dir = '/draft';
      file._empty = true;
      file.navigation = false;
    }
  })
})
```
It's obviously a dirty hack, but it works well enough for my purposes.

Hope this helps! Thank you for your time.

**Comment by preethamrn** [comment]
The workarounds and analysis done by previous commenters was great. If there's no plan to work on fixing this maybe the documentation should be updated? It would definitely have saved me a lot of headache since it's a bit misleading as it is right now.

**Comment by mynameisjeffe** [comment]
Workaround I use is this
 
https://content.nuxt.com/get-started/configuration#ignores
`export default defineNuxtConfig({
  content: {
    ignores: [
      'hidden',        // any file or folder that contains the word "hidden"
    ]
  }
})
`
I then append "-hidden" for any contents that are still in draft state.

**Comment by XStarlink** [comment]
> Workaround I use is this
> 
> https://content.nuxt.com/get-started/configuration#ignores
> 
> ```
> export default defineNuxtConfig({
>   content: {
>     ignores: [
>       'hidden',        // any file or folder that contains the word "hidden"
>     ]
>   }
> })
> ```
> 
> I then append "-hidden" for any contents that are still in draft state.

Instead of doing this Workaround, you can prefix the names of your files inside the 'content' folder with a dot, e.g. '.pageName.md', and it will be ignored and therefore not displayed.
It works for folders too.

But I agree that "draft: true" in the Front-matter should also work as described in the doc here: https://content.nuxt.com/usage/markdown#front-matter

**Comment by zrooda** [comment]
This was an unpleasant surprise when we deployed to prod, but my mistake blindly trusting the documentation. Either way the docs should remove mentioning this feature until it actually works.

**Comment by HADB** [comment]
@farnabaz Hi, this issue has not been completely resolved in #2648.

Adding `.draft` to the filename of markdown file works for excluding it, but setting `draft: true` in the markdown file as document saying does not take effect.


<img width="231" alt="iShot_2024-08-01_16 09 40" src="https://github.com/user-attachments/assets/2f9cbd2a-3c6b-4f67-9633-ba82d2591ce1">


`draft` is `true` but `_draft` is `false`.


<img width="581" alt="iShot_2024-08-01_16 09 27" src="https://github.com/user-attachments/assets/83345047-d953-4f97-802b-5813a9c6a6e6">
